src/frontend/screens/Audio/AudioPlaybackScreen.tsx

```diff
@@ -0,0 +1,214 @@
+    isOpen: isOpenShareSheet,
+    openSheet: openShareSheet,
+    closeSheet: closeShareSheet,
+  } = useBottomSheetModal({
+    openOnMount: false,
+  });
+
+  const navigation = useNavigationFromRoot();
+  const {recordingUri, previewOnly} = params.route.params;
+  const observation = useDraftObservation();
+
+  const {
+    isReady,
+    isPlaying,
+    currentPosition,
+    duration,
+    startPlayback,
+    stopPlayback,
+  } = useAudioPlayback(recordingUri);
+
+  useFocusEffect(() => {
+    navigation.setOptions({
+      // eslint-disable-next-line react/no-unstable-nested-components -- it's supported way to do it like that
+      headerLeft: () => (
+        <MaterialIcon
+          name="close"
+          size={25}
+          color={WHITE}
+          onPress={() => {
+            if (previewOnly) {
+              navigation.goBack();
+            } else {
+              observation.addAudioRecording({
+                duration,
+                uri: recordingUri,
+                createdAt: new Date().getTime(),
+              });
+              openSuccessSheet();
+            }
+          }}
+        />
+      ),
+    });
+  });
+
+  const handleControlButtonPress = async () => {
+    isPlaying ? await stopPlayback() : await startPlayback();
+  };
+
+  const formattedDuration = isReady
+    ? Duration.fromMillis(duration).toFormat('mm:ss')
+    : '00:00';
+
+  const elapsedTimeValue = useDerivedValue(() => {
+    return withTiming(currentPosition, {duration: 500});
+  }, [currentPosition]);
+
+  return (
+    <>
+      <StatusBar style="light" />
+      <SafeAreaView style={styles.container}>
+        <AnimatedTimer elapsedTime={elapsedTimeValue} />
+        <AnimatedProgressBar
+          isReady={isReady}
+          elapsed={elapsedTimeValue}
+          duration={duration}
+        />
+        <Text style={styles.textStyle}>Total length: {formattedDuration}</Text>
+        <View style={styles.bottomBar}>
+          <Pressable onPress={openDeleteBottomSheet}>
+            <MaterialIcons size={35} name="delete" color={WHITE} />
+          </Pressable>
+          <Pressable
+            disabled={!isReady}
+            onPress={handleControlButtonPress}
+            style={styles.buttonWrapperStyle}>
+            {isPlaying ? (
+              <View style={styles.buttonStopStyle} />
+            ) : (
+              <PlayArrow />
+            )}
+          </Pressable>
+          <Pressable onPress={openShareSheet}>
+            <MaterialIcons size={35} name="share" color={WHITE} />
+          </Pressable>
+        </View>
+        <AudioRecordingDeleteBottomSheet
+          isOpen={isOpenDeleteBottomSheet}
+          closeSheet={closeDeleteBottomSheet}
+          sheetRef={deleteAudioSheetRef}
+          recordingUri={recordingUri}
+          previewOnly={previewOnly}
+        />
+        <AudioRecordingSuccessBottomSheet
+          isOpen={isOpenSuccessSheet}
+          sheetRef={successSheetRef}
+        />
```

Comment:
After we add `SuccessBottomSheet` it should look like that.

```suggestion
         <SuccessBottomSheet
          isOpen={isOpenSuccessSheet}
          sheetRef={successSheetRef}
          icon={<SuccessIcon />}
          title={formatMessage(m.successTitle)}
          description={
            <FormattedMessage
              {...m.successDescription}
              values={{
                audioRecording: 'Audio Recording',
                // eslint-disable-next-line react/no-unstable-nested-components
                bold: message => <Text style={styles.textBold}>{message}</Text>,
              }}
            />
          }
          handleOutlinedButton={() => navigation.navigate('ObservationEdit')}
          outlinedButtonText={formatMessage(m.returnToEditorButtonText)}
          handleFillButton={() =>
            navigation.navigate('Audio', {screen: 'PrepareRecording'})
          }
          fillButtonText={formatMessage(m.recordAnotherButtonText)}
        />
```